### ISSUE TITLE
Heisenbug: trip _gateway panic sometimes

### ISSUE THREAD

**Issue by crrodriguez** [fujiapple852/trippy#892]
```
#trip _gateway
thread 'main' panicked at /rustc/3cdd004e55c869faa2b7b25efd3becf50346e7d6/library/core/src/cmp.rs:879:9:
assertion failed: min <= max
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
Unfortunately it cannot be reproduced the second time ..it happens once in a while.. I'l trry to get a backtrace ..

**Comment by fujiapple852** [comment]
Thanks for reporting @crrodriguez.

please also try enabling verbose logging with -v which may help give more context around the failure.

**Comment by fujiapple852** [comment]
The assertion failure line from the reported error is from the `clamp` function:

https://github.com/rust-lang/rust/blob/3cdd004e55c869faa2b7b25efd3becf50346e7d6/library/core/src/cmp.rs#L879

**Comment by fujiapple852** [comment]
@crrodriguez I can reproduce this failure by pressing `}` (default binding for `expand-hosts-max`), can you confirm this is what you are doing when it fails?

The bug here is in the `TuiApp::max_hosts` function:

https://github.com/fujiapple852/trippy/blob/c65a57e1a7781b0b3bcd2df74c8d9b901a9ca0a1/src/frontend/tui_app.rs#L388-L398

If there are no hops reported or none of the reported hops contain any addresses, then the function returns `0` (i.e. `u8::default()` from `unwrap_or_default()`).  This will cause `TuiApp::expand_hosts_max` to set `max_addrs` as `Some(0)`:

https://github.com/fujiapple852/trippy/blob/c65a57e1a7781b0b3bcd2df74c8d9b901a9ca0a1/src/frontend/tui_app.rs#L380-L382

With that in place, we will now get a call to `clamp(1, 0)` which will lead to the assertion failure observed:

https://github.com/fujiapple852/trippy/blob/c65a57e1a7781b0b3bcd2df74c8d9b901a9ca0a1/src/frontend/render/table.rs#L254

The fix is to ensure that in such cases `max_addrs` is set to be `None`.

Given you are running `trip _gateway` i'm assuming that means `_gateway` is the first (and only) hop and so it would seem likely that if the `_gateway` hop does not respond to the IMCP probe then we would indeed end up in a situation where none of the hops contain any addresses and would therefore lead to this bug.

**Comment by fujiapple852** [comment]
Fixed in #893 which will go into the 0.10.0 release.

**Comment by crrodriguez** [comment]
> @crrodriguez I can reproduce this failure by pressing `}` (default binding for `expand-hosts-max`), can you confirm this is what you are doing when it fails?

Yes, See in my spanish language keyboard that key is annoyingly too close to enter and sometimes both get pressed.. 

I updated to https://github.com/fujiapple852/trippy.git#58dc9b58) and the problem appears to be gone.

Thanks.

**Comment by c-git** [comment]
Thank you very much for testing and confirming that works @crrodriguez

**Comment by fujiapple852** [comment]
I merged #893 so you can retry from the latest master (f07d58f768f3cdd6b9179775b575059ef4f8512c) if you wish.

**Comment by crrodriguez** [comment]
> I merged #893 so you can retry from the latest master ([f07d58f](https://github.com/fujiapple852/trippy/commit/f07d58f768f3cdd6b9179775b575059ef4f8512c)) if you wish.

, I did try it and the issue is gone. thanks.!
